frappe/desk/search.py

```diff
@@ -171,6 +171,7 @@ def search_widget(doctype, txt, query=None, searchfield=None, start=0,
 
 			if doctype in UNTRANSLATED_DOCTYPES:
 				values = tuple([v for v in list(values) if re.search(re.escape(txt)+".*", (_(v.name) if as_dict else _(v[0])), re.IGNORECASE)])
+				values = tuple(sorted(values, key= lambda x: ((_(x.name) if as_dict else _(x[0])).lower().startswith(txt.lower()) == False, (_(x.name) if as_dict else _(x[0])))))
```

Comment:
Too convoluted to fit in a single line, and not readable. Multiple statements in a single line are not preferred ([PEP8](https://www.python.org/dev/peps/pep-0008/) [E701](https://www.flake8rules.com/rules/E701.html)). 

Also, try handing operations via L173 and 174 during a single iteration over the values. The existing logic can be simplified too.

Reply:
okay, thanks for the review. I will do the necessary changes.

Reply:
Maybe I wasn't clear enough earlier. Separate the list comprehension, use an appropriately named function instead of the lambda function. Even though it's styled across multiple lines, it's still a single statement expression, which is still slightly convoluted.

tldr; it's not easy to understand what's happening at first glance. Maybe add a comment that says 'why' and 'what' are you doing this for

Reply:
@gavindsouza , sorry I thought that introducing a new function would be unnecessary that's why I used the lambda function. I have now done the changes as you asked.